# Question:  Should I attend CCA first or go directly to the Colorado School of Mines if I want to minimize my debt after completing a bachelors degree.?

The random variable that I am measuring is the debt accumulated after completing the Masters Degree.
One trial of this simulation would involve a random student either complete a 2-year degree through CCA first or go through Mines.

In [1]:
import random as rd
from collections import Counter
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sb
import statistics

In [2]:
df = pd.read_csv("GradData.csv")

In [3]:
# Run the simulator 100 times
passMines = []
passCCA = []
for i in range(100):
    flip = rd.randint(1, 2)
    if(flip == 1):
        chanceM = rd.randint(1, 100)
        passMines.append(chanceM)
    else:
        chanceC = rd.randint(1, 100)
        passCCA.append(chanceC)

In [4]:
costGMines = []
costFMines = []
annualCostM = df.loc[df["School"] == "Mines", "AvgAnCost"].values[0]
gradPercentM = df.loc[df["School"] == "Mines", "GradPercent"].values[0]
for x in passMines:
    if(x <= gradPercentM):
        costGMines.append(annualCostM * 4)
    else:
        yrs = rd.randint(0, 3)
        costFMines.append(annualCostM * yrs)

In [14]:
costGCCA = []
costFCCA = []
annualCostC = df.loc[df["School"] == "CCA", "AvgAnCost"].values[0]
gradPercentC = df.loc[df["School"] == "CCA", "GradPercent"].values[0]
for x in passCCA:
    if(x <= gradPercentC):
        costGCCA.append(annualCostC * 2)
    else:
        yrs = rd.randint(0, 1)
        costFCCA.append(annualCostC * yrs)

costGCCAm = []
costGCCAM = []
for i, x in enumerate(costGCCA):
    rand = rd.randint(1, 100)
    if rand <= gradPercentM:
        costGCCAM.append(x + (annualCostM * 2))
    else:
        yrs = rd.randint(0, 1)
        costGCCAm.append(x + (annualCostM * yrs))

In [15]:
import numpy as np

# Convert lists to numpy arrays to use the mean() method
# Assuming costGCCA, costFCCA, costGMines, costFMines are lists
aGCCAM = np.mean(costGCCAM)
aGCCAm = np.mean(costGCCAm)
aFCCA = np.mean(costFCCA)
aGMines = np.mean(costGMines)
aFMines = np.mean(costFMines)

print(aGCCAM, '\n', aGCCAm, '\n', aFCCA, '\n', aGMines, '\n', aFMines)

73412.0 
 14932.0 
 4035.675675675676 
 116960.0 
 29240.0


In conclusion, the cost of attending CCA first is 73,412 dollars, which is much cheaper than attending Mines for all 4 years, which costs 116,960 dollars.